### DISCUSSION TITLE
Updating dependency of private go module

### DISCUSSION THREAD

**Discussion by gursharanbains** [renovatebot/renovate#13710] [Request Help]
Hi,

We have two private go module repos on Github, one of them depends on the other. I am trying to configure Renovate App so that Renovate can track all commits made to the dev (default) branch of the dependency and use the pseudo-version to update go.mod of the first repo. I have installed Renovate App (and disabled it) for the dependency repo so that the bot has permission to access it. However although further commits have been made to the dev branch of the dependency no updates are showing in the first repo's Configure Renovate PR or in the logs.

Logs do not show any errors but there is an attempt to access "https://proxy.golang.org/" for the dependency which I assume is a routine check? There is another debug statement after it

```
"GitHub 404
{
  "url": "https://api.github.com/repos/{org-name}/{repo-name}/tags?per_page=100"
}
```
And after that a message "Failed to look up dependency".

Could anyone help me with this, and let me know is it possible to track the dev branch of the dependency and how I can accomplish it?

**Comment by rarkins** [comment]
I'm not sure it's possible to do what you need.

But if you see a 404 and you'd expect that to work, then it would be good to troubleshoot that and work out why first.

**Reply by gursharanbains** [reply]
I am actually not sure why it is not working. I suspect it is something to do with access token, but I assumed that adding Renovate to the dependency repo would ensure that Renovate has all necessary permissions in its token? Or is there a configuration I need to add in hostRules perhaps?

**Reply by rarkins** [reply]
How are you running Renovate? e.g. self-hosting?

**Reply by gursharanbains** [reply]
We are using the Github app.

**Reply by rarkins** [reply]
The ephemeral token generated in the app should be able to access all repositories within the same org, as long as they are both private. So the 404 seems unexpected.

**Reply by gursharanbains** [reply]
Could it have to do with the fact that I haven't merged the Configure Renovate PR yet for the first repo? I had held off on it because I had assumed I would be able to check and see the private dependency showing up in the PR first before I merged it.

**Reply by rarkins** [reply]
No, that won't be the cause

**Reply by gursharanbains** [reply]
Checking after a few days, I saw that the proxy.golang.org and 404 errors were not showing anymore in the logs. Just to be sure, I closed the Configure Renovate PR so that it could be recreated, and those errors do not show now. But I still don't see any updates for the private repo

**Reply by rarkins** [reply]
The 404 from api.github.com still indicates that there's a credentials problem somehow

**Reply by gursharanbains** [reply]
I am not seeing any 404 errors now related to the private repo. But it does not show any updates for the private repo. Could this have to do with the fact that I want to track commits made to the dev branch of the private repo and apply updates using its new golang pseudo-version? Is there a way to configure Renovate to do this?

**Reply by rarkins** [reply]
Can you reproduce this scenario in a public repo?

**Comment by PhilipAbed** [comment]
i have tried to add my private repo in the host rules, and got failure too, that's my configuration with `encrypted` token plus my logs
![image](https://user-images.githubusercontent.com/42116482/163198402-52d6c69a-9082-4be1-960e-24df6b64a4fb.png)
![image](https://user-images.githubusercontent.com/42116482/163198610-dfdd7bdc-2676-49b3-aae9-90685383c608.png)

but it's also failing, im not sure how to keep my private packages updated too

**Reply by viceice** [reply]
you need a second rule with `https://api.github.com/repos/<org>/<repo/`, as we do the github tags via github api.

it probably needs to be `hostType=github-tags`

**Reply by PhilipAbed** [reply]
yeah i tried that too but still fails same issue 404
![image](https://user-images.githubusercontent.com/42116482/163203600-b513cf3d-add6-4380-80f5-f0322fe46106.png)

**Reply by viceice** [reply]
I think you probably need to debug that yourself. I would recommend to remove `/tags...` from `matchHost`, as renovate can change the appended args while paging

**Reply by viceice** [reply]
check the request summary at the end of the repo run, you should see all requests, maybe some are not matched to your hostrules

**Reply by viceice** [reply]
THis PR will probably help to find failed requests:
- https://github.com/renovatebot/renovate/pull/15104

**Reply by rarkins** [reply]
I've seen that the Link: header for some github paginated requests can link to `api.github.com/repositories/<id>/` which would then mean our hostRules don't match. Been meaning to look into it.

**Reply by PhilipAbed** [reply]
the user successfully did that but he went through another flow using GOPROXY and he set the password in the URL inside go proxy, im not sure how to reach that flow though `releases-goproxy.ts` , i put a debug there but code is not reaching, im trying to find out

**Reply by rarkins** [reply]
I didn't think that private repos and GOOROXY are compatible - you'd need the proxy to fail and then fall back to "direct"

**Reply by PhilipAbed** [reply]
maybe u are right, i'm thinking about this in the wrong way, maybe i need to turn a private repo into a GO Registry and then put it in go proxy then it might work not sure yet.
